packages/core/src/ExtractValue.ts

```diff
@@ -0,0 +1,184 @@
+import {
+	INode,
+	INodeParameters,
+	INodeProperties,
+	INodePropertyCollection,
+	INodePropertyOptions,
+	INodeType,
+	NodeOperationError,
+	NodeParameterValueType,
+	NodeHelpers,
+} from 'n8n-workflow';
+
+function findPropertyFromParameterName(
+	parameterName: string,
+	nodeType: INodeType,
+	node: INode,
+	nodeParameters: INodeParameters,
+): INodePropertyOptions | INodeProperties | INodePropertyCollection {
+	let property: INodePropertyOptions | INodeProperties | INodePropertyCollection | undefined;
+	const paramParts = parameterName.split('.');
+	let currentParamPath = '';
+
+	const findProp = (
+		name: string,
+		options: Array<INodePropertyOptions | INodeProperties | INodePropertyCollection>,
+	): INodePropertyOptions | INodeProperties | INodePropertyCollection | undefined => {
+		return options.find(
+			(i) =>
+				i.name === name &&
+				NodeHelpers.displayParameterPath(nodeParameters, i, currentParamPath, node),
+		);
+	};
+
+	// eslint-disable-next-line no-restricted-syntax
+	for (const p of paramParts) {
+		const param = p.split('[')[0];
+		if (!property) {
+			property = findProp(param, nodeType.description.properties);
+		} else if ('options' in property && property.options) {
+			property = findProp(param, property.options);
+			currentParamPath += `.${param}`;
+		} else if ('values' in property) {
+			property = findProp(param, property.values);
+			currentParamPath += `.${param}`;
+		} else {
+			throw new Error(`Couldn't not find property "${parameterName}"`);
+		}
+		if (!property) {
+			throw new Error(`Couldn't not find property "${parameterName}"`);
+		}
+	}
+	if (!property) {
+		throw new Error(`Couldn't not find property "${parameterName}"`);
+	}
```

Comment:
would this stop execution? if property is set that cannot be found? does it need to?

Reply:
It will stop execution. This probably shouldn't happen except when developing a node.

Reply:
yes, it's a bit of an edge case.. I would lean towards not breaking execution.. but I will defer to you on this one..